src/js/components/alert.js

```diff
@@ -2,46 +2,61 @@
  * Copyright (C) 2018 The Trustees of Indiana University
  * SPDX-License-Identifier: BSD-3-Clause
  */
+import dispatchCustomEvent from '../utilities/dispatchCustomEvent';
+import { isNode } from '../utilities/domHelpers';
 
 export default class Alert {
+  constructor(element) {
 
-  constructor(options) {
-    this.handleClick = this.handleClick.bind(this);
+    // Instance properties
+    this.element = element;
+    this.closeAttribute = 'data-alert-close';
+    this.closeSelector = `[${this.closeAttribute}]`;
 
-    this.init(options);
-  }
-
-  init(options) {
-    document.getElementById(options.id).closest('[aria-labelledby]').querySelector('[data-alert-close]').addEventListener('click', () => { this.handleClick(options.id); }, false);
-  }
-
-  handleClick(id) {
-    const dismissButton = event.target.closest('[data-alert-close]');
+    // bind methods
+    this._handleClick = this._handleClick.bind(this);
 
-    // If the target wasn't the dismiss button bail.
-    if (!dismissButton) return;
+    // Check to make sure that a DOM element was passed in for initialization
+    if (!isNode(this.element)) {
+      throw new TypeError(
+        'A DOM element should be passed as the first argument to initialize the sidenav'
+      );
+    }
 
-    this.dismissAlert(id);
+    this.init();
   }
 
-  dismissAlert(id, callback) {
-    let alert = document.querySelector('[aria-labelledby="' + id + '"]');
-
-    if (!alert) {
-      alert = document.getElementById(id);
-    }
+  _handleClick() {
+    this.dismiss();
+  }
 
-    if (!alert) {
-      throw new Error(
-        'Could not find an alert with the id of ' + id + ' to dismiss.'
-      );
-    }
+  /**
```

Comment:
This comment should live inside the `dismiss()` method body next to the `dismissEvent` statement.